File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_Giustificativi/Models/par-giustificativi-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";


export class Par_GiustificativiModel {

  public id!: number;
  public idAz_Anagrafica!: number;
  public descrizione!: string;
  public codice!: string;

  public backgroundColor!: string;
  public textColor!: string;
  public tipoInput!: JustTipoInput;
  public idCausale?: number;
  public segno!: SignWithNeutral;
  public visualizzaInPianoFerie!: boolean;
  public tipoContatore: TipoContatore = TipoContatore.NoContatore;

}

export enum SignWithNeutral {
  Down = -1,
  Neutral = 0,
  Up = 1
}

export enum JustTipoInput {
  InteraGiornate,
  Intervallo,
  Tutti
}

export enum TipoContatore {
  NoContatore        = 0,
  Contatore          = 1,
  ContatoreConAvviso = 2,
  ContatoreConBlocco = 3
}


export class Par_GiustificativiInModel {
  
}
export class Par_GiustificativiOutModel extends ModelResult {

  public par_Giustificativi: Par_GiustificativiModel[];

}



export class Par_GiustificativiGetInModel {
  public id: number;
}
export class Par_GiustificativiGetOutModel extends ModelResult {
  public par_Giustificativi: Par_GiustificativiModel;
}
export class Par_GiustificativiPutInModel {
  public par_Giustificativi: Par_GiustificativiModel;
}
export class Par_GiustificativiPutOutModel extends ModelResult {
  public par_Giustificativi: Par_GiustificativiModel;
}

export class Par_Giustificativi_DeleteInModel {
  id: number;
}
export class Par_Giustificativi_DeleteOutModel extends ModelResult {
}

File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_GiustificativiService/Models/Par_GiustificativiModel.cs
using nvxapp.server.data.Entities.Tenant;
using nvxapp.server.data.Extensions;
using nvxapp.server.service.ClientServer_Service.Infrastructure.Account.Models;
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_GiustificativiService.Models
{
    public class Par_GiustificativiModel
    {
        public int Id { get; set; }
        public int IdAz_Anagrafica { get; set; }
        public string? Descrizione { get; set; }
        public string? Codice { get; set; }
        public string? BackgroundColor { get; set; }
        public string? TextColor { get; set; }
        public JustTipoInput TipoInput { get; set; }
        public int? IdCausale { get; set; }
        public SignWithNeutral Segno { get; set; }
        public bool VisualizzaInPianoFerie { get; set; }
    }


    public class Par_GiustificativiInModel
    {

    }
    public class Par_GiustificativiOutModel : ModelResult 
    {
        public List<Par_GiustificativiModel> Par_Giustificativi { get; set; } = new List<Par_GiustificativiModel>();



        public Par_GiustificativiOutModel() 
        {
        
        }
    }


    public class Par_GiustificativiGetInModel
    {
        public int Id { get; set; } = 0;
    }
    public class Par_GiustificativiGetOutModel : ModelResult
    {
        public Par_GiustificativiModel Par_Giustificativi { get; set; } = new Par_GiustificativiModel();
    }
    public class Par_GiustificativiPutInModel : ModelResult
    {
        public Par_GiustificativiModel Par_Giustificativi { get; set; } = new Par_GiustificativiModel();
    }
    public class Par_GiustificativiPutOutModel : ModelResult
    {
        public Par_GiustificativiModel Par_Giustificativi { get; set; } = new Par_GiustificativiModel();
    }

    public class Par_Giustificativi_DeleteInModel
    {
        public int Id { get; set; }
    }
    public class Par_Giustificativi_DeleteOutModel : ModelResult
    {
    }
}
